packages/bundle/src/createCognitiveServicesSpeechServicesPonyfillFactory.js

```diff
@@ -0,0 +1,39 @@
+import createPonyfill from 'web-speech-cognitive-services/lib/SpeechServices';
+
+export default async function ({
+  authorizationToken,
+  region,
+  subscriptionKey,
+  textNormalization
+}) {
+  const ponyfill = await createPonyfill({
+    authorizationToken,
+    region,
+    subscriptionKey,
+    textNormalization
+  });
+
+  const {
+    SpeechGrammarList,
+    speechSynthesis,
+    SpeechSynthesisUtterance
+  } = ponyfill;
+
+  return ({ referenceGrammarID }) => ({
+    SpeechGrammarList,
+    SpeechRecognition: injectReferenceGrammarID(ponyfill, referenceGrammarID),
+    speechSynthesis,
+    SpeechSynthesisUtterance
+  });
+}
+
+function injectReferenceGrammarID({ SpeechGrammarList, SpeechRecognition }, referenceGrammarID) {
```

Comment:
We inject the "reference grammar ID" because this property is not a standard in W3C. Instead of polluting the spec with hidden properties, we prefer to inject it whenever `start` is called. Since `start` isn't calling very often (once every few seconds, at most), this pattern won't introduce observable burden.